src/pages/BuySell/SelectLayout.js

```diff
@@ -0,0 +1,222 @@
+import { React, useState } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import styled from 'styled-components';
+import { flexBox } from '../../styles/mixin';
+
+const SIZE_BTN = [
+  { size: 230 },
+  { size: 240 },
+  { size: 250 },
+  { size: 260 },
+  { size: 270 },
+  { size: 280 },
+];
+
+const SelectLayout = ({ type }) => {
+  const [selectSize, setSelectSize] = useState();
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const navigate = useNavigate();
+
+  const onSizeClick = size => {
+    setSelectSize(size);
+    searchParams.set('size', size);
+    setSearchParams(searchParams);
+  };
+  return (
+    <BuyBackGround>
+      <Title>{type === 'sell' ? '판매하기' : '구매하기'}</Title>
+      <BuyWrapper>
+        <BuyBox>
+          <ItemInfoBox>
+            <ItemImg
+              src="http://placeimg.com/640/640/people"
+              alt="제품 이미지"
+              width="80px"
+            />
+            <ItemInfo>
+              <ItemCode>DZ5485-612</ItemCode>
+              <ItemEnglishName>
+                Jordan 1 Retro High OG Chicago 2022
+              </ItemEnglishName>
+              <ItemKoreanName>조던 1 레트로 하이 OG 시카고 2022</ItemKoreanName>
+            </ItemInfo>
+          </ItemInfoBox>
+
+          <ItemSelectBox>
+            <SelectList>
+              {SIZE_BTN.map(({ size }) => {
+                return (
+                  <SelectItem
+                    onClick={() => onSizeClick(size)}
+                    className={size === selectSize && 'active'}
+                    key={size}
+                  >
+                    <SizeBtn>
+                      <Size>{size}</Size>
+                      <Price type={type}>123,000</Price>
+                    </SizeBtn>
+                  </SelectItem>
+                );
+              })}
+            </SelectList>
+          </ItemSelectBox>
+          {selectSize && (
+            <BtnArea>
+              <BtnBox
+                onClick={() => {
+                  navigate(`/${type}?${searchParams.toString()}`);
+                }}
```

Comment:
navigate 비하인더씬으로 state 보내는 리뷰가 사라졌군요
queryString으로 만들어 주신 것도 좋은 방법 인 것 같습니다.

그런데 혹시 이건 어떠신가요?
```suggestion
                onClick={() => {
                  navigate(`/${type}?size=${selectedSize}`);
                }}
```

Reply:
감사합니다!! 
남겨주신 리뷰 확인하고 useLocation 방식과 searchParams 방식 둘다 사용해 보았는데 
현재 상황에서는 어떤 방식을 사용하는게 더 좋을까요??

Reply:
searchParam 
장점 : 편하다..
단점 : 사용자가 쉽게 값을 임의로 변경 할 수 있다.

location (비하인더씬으로 state보내주기)
장점 : 사용자가 쉽게 값을 임의로 변경 할 수 없다. (물론 어렵게는 가능합니다)
단점 : state가 안들어왔을 때에 추가적인 핸들링이 필요하다
         ex) 다시 뒤로가기를 해주던지 (navigate(-1)) 버튼을 클릭하는 페이지로 이동시켜줘야 한다. (buy, sell 쪽)

위 장단점을 비교 하셔서 적용하면 더 좋을 것 같습니다!

이 상황에선 만약 저라면 비하인더씬을 사용할 것 같습니다.